src/niscope/system_tests/test_system_niscope.py

```diff
@@ -93,10 +104,19 @@ def test_fetch_binary16_into(session):
     session.configure_horizontal_timing(50000000, test_record_length, 50.0, 1, True)
     with session.initiate():
         wfm_infos = session.channels[test_channels].fetch_into(wfm=wfm)
+
     for sample in wfm:
         assert not math.isnan(sample)
     assert len(wfm_infos) == test_num_channels
 
+    if sys.version_info.major >= 3:
+        # Only python 3 will have the record memory view in the wfm_info
+        for i in range(len(wfm_infos)):
+            record_wfm = wfm_infos[i].wfm
+            assert len(record_wfm) == test_record_length
+            for j in range(len(record_wfm)):
+                assert record_wfm[j] == wfm[i * test_record_length + j]
+
```

Comment:
else

ensure that there is no waveform data in the waveform info object?

Reply:
Added